decidim-proposals/app/controllers/decidim/proposals/proposals_controller.rb

```diff
@@ -65,7 +65,9 @@ def create
       private
 
       def order
-        @order = params[:order] || "random"
+        return @order if @order
+        @order = "random" if current_settings.votes_enabled? && current_settings.votes_hidden? && params[:order] == "most_voted"
+        @order ||= params[:order] || "random"
```

Comment:
If votes are enabled and hidden, I don't want anyone to be able to hack the URL to order proposals by most-voted. That's why first I set a variable for this case. If the conditions do not match, then I use the old behaviour.